### PR TITLE
Run golangci in server CI

### PR DESCRIPTION
**Pull request by agnivade**
Also reordered the checks in check-style target to run vet check
first which has far better error messages in case of unsynced branch failures.
```release-note
NONE
```

### CODE REVIEW

**Review by isacikgoz — COMMENTED** [review]
We can have a separate job and use https://github.com/golangci/golangci-lint-action instead. My initial plan was to do so, but seems like I missed that out.

**Comment by agnivade** [comment]
@isacikgoz -Thanks, PTAL.

**Review by isacikgoz — APPROVED** [review]
Nice!

File: server/Makefile [hunk 1]
```diff
@@ -291,7 +291,7 @@ else
 endif
 
 golangci-lint: ## Run golangci-lint on codebase
-	@# Keep the version in sync with the command in .circleci/config.yml
+	@# Keep the version in sync with the command in .github/workflows.yml#L99
```
Comment (agarciamontoro) [inline server/Makefile:294]: This file does not exist, what should it say instead?

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
The new action looks good! Just a comment on a comment :)

File: server/Makefile (hunk 1 above)
Comment (agnivade) [reply server/Makefile:294]: lol, it should say .github/workflows/ci.yml#L99

Comment (agarciamontoro) [reply server/Makefile:294]: But line 99 in that file is just `cache-dependency-path: server/go.sum` :joy:

Comment (agnivade) [reply server/Makefile:294]: Yeah I was switching between branches. Got that messed up.

**Review by agarciamontoro — APPROVED** [review]

**Pull request merged**